src/Products/Comparison/Service/ComparisonServiceImpl.cs

```diff
@@ -110,250 +97,175 @@ public bool CheckFiles(CompareRequest files)
-                }
-            }
 
-            CompareResultResponse compareResultResponse = GetCompareResultResponse(compareResult, changes, extension);
+            CompareResultResponse compareResultResponse = GetCompareResultResponse(changes, resultGuid);
             compareResultResponse.SetExtension(extension);
             return compareResultResponse;
         }
 
-        private ICompareResult CompareFiles(CompareRequest compareRequest, bool detectDeletions)
+        private static Comparer CompareFiles(CompareRequest compareRequest, string resultGuid)
         {
             string firstPath = compareRequest.guids[0].GetGuid();
-            ICompareResult compareResult;
+            string secondPath = compareRequest.guids[1].GetGuid();
+
             // create new comparer
-            Comparer comparer = new Comparer();
-            // create setting for comparing
-            ComparisonSettings settings = new ComparisonSettings();
-            settings.ShowDeletedContent = false;
-            settings.StyleChangeDetection = true;
-            settings.CalculateComponentCoordinates = true;
-            // compare two documents            
-            // if we need to detect deletions we should revers compared files to get correct coordinates
-            if (detectDeletions)
-            {
-                compareResult = comparer.Compare(compareRequest.guids[1].GetGuid(),
-                   compareRequest.guids[1].GetPassword(),
-                   firstPath,
-                   compareRequest.guids[0].GetPassword(),
-                   settings);
-            }
-            else
+            Comparer comparer = new Comparer(firstPath, GetLoadOptions(compareRequest.guids[0].GetPassword()));
+
+            comparer.Add(secondPath, GetLoadOptions(compareRequest.guids[1].GetPassword()));
+            CompareOptions compareOptions = new CompareOptions { CalculateCoordinates = true };
+
+            if (Path.GetExtension(resultGuid) == ".pdf")
             {
-                compareResult = comparer.Compare(firstPath,
-                    compareRequest.guids[0].GetPassword(),
-                    compareRequest.guids[1].GetGuid(),
-                    compareRequest.guids[1].GetPassword(),
-                    settings);
+                compareOptions.DetalisationLevel = DetalisationLevel.High;
             }
-            if (compareResult == null)
+
+            using (FileStream outputStream = File.Create(Path.Combine(resultGuid)))
             {
-                throw new Exception("Something went wrong. We've got null result.");
+                comparer.Compare(outputStream, compareOptions);
             }
-            return compareResult;
+
+            return comparer;
         }
 
-        private CompareResultResponse GetCompareResultResponse(ICompareResult compareResult, ChangeInfo[] changes, string ext)
+        private CompareResultResponse GetCompareResultResponse(ChangeInfo[] changes, string resultGuid)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetCompareResultResponse' a static method.](https://app.codacy.com/gh/groupdocs-comparison/GroupDocs.Comparison-for-.NET-WebForms/pullRequest?prid=5451797)